[IMP] tests: add test to detect deprecated field values (digits_compute, select)

tomeyro opened the pull request:
Referring issue: https://github.com/OCA/pylint-odoo/issues/84

moylop260 commented on pylint_odoo/checkers/no_modules.py:
```diff
@@ -186,6 +186,11 @@
         settings.DESC_DFLT
     ),
+    'W%d11' % settings.BASE_NOMODULE_ID: (
+        'Parameter "%s" is no longer supported. Use "%s" instead.',
```
`field parameter...`

tomeyro replied:
👍

moylop260 commented on pylint_odoo/checkers/no_modules.py:
```diff
@@ -243,6 +248,11 @@
+}
+
 
 class NoModuleChecker(BaseChecker):
```
Use a options in order to add this values from pylintrc file.
Looking for similar examples with other `DFLT` variables used from options

tomeyro replied:
I didn't find a way to parse the options as dictionaries so I refactored the variable a bit and am using a regex to check for the deprecated field params. Please take a look.

tomeyro commented:
@moylop260 

I refactored the check. Now it parses the value for `DFTL_DEPRECATED_FIELD_PARAMETERS` on the `open` function, running only once per test. It didn't work in `__init__` because in some other process the options are parsed again and the transformation from `old:new` to `{'old': 'new'}` is lost, but `open` is called just before the tests start.

Also, I changed the name of the message from `deprecated-field-parameters` to `renamed-field-parameters` to keep the previous message available for when a check for a removed field param is needed, and that check will be able to use the same `DFTL_DEPRECATED_FIELD_PARAMETERS` var because it parses `old:new,old_param` to `{'old': 'new', 'old_param': None}`.

moylop260 commented on pylint_odoo/checkers/no_modules.py:
```diff
@@ -327,10 +348,29 @@ class NoModuleChecker(BaseChecker):
+        self.config.deprecated_field_parameters = \
+            self.colon_list_to_dict(self.config.deprecated_field_parameters)
+
+    def colon_list_to_dict(self, colon_list=None):
```
`colon_list` arg should be required then you could remove a default value None

moylop260 commented on pylint_odoo/checkers/no_modules.py:
```diff
@@ -327,10 +348,29 @@ class NoModuleChecker(BaseChecker):
+            {'colon': 'list', 'just_key': None}
+        """
+        tmp_dict = {}
+        for item in colon_list if isinstance(colon_list, list) else []:
```
self.config.deprecated_field_parameters returns even a list
Then we could avoid a extra validation here.
If someone use wrong this method then don't read correctly the docstring

moylop260 commented on pylint_odoo/checkers/no_modules.py:
```diff
@@ -327,10 +348,29 @@ class NoModuleChecker(BaseChecker):
+            self.colon_list_to_dict(self.config.deprecated_field_parameters)
+
+    def colon_list_to_dict(self, colon_list=None):
+        """ \
```
```diff
- """ \
+ """ Converts a colon list to a dictionary
```

moylop260 commented on pylint_odoo/checkers/no_modules.py:
```diff
@@ -327,10 +348,29 @@ class NoModuleChecker(BaseChecker):
+        tmp_dict = {}
+        for item in colon_list if isinstance(colon_list, list) else []:
+            prop = item.split(":")
+            tmp_dict[prop[0]] = prop[1] if len(prop) > 1 else None
```
What about use `dict([item.split(":") for item in colon_list])`?
![screen shot 2016-12-23 at 3 27 47 pm](https://cloud.githubusercontent.com/assets/6644187/21462630/daa080fc-c923-11e6-974f-f5ed8912b7ea.png)

moylop260 replied:
We don't need a extra validation here since that we are using a good help from `option` definition

tomeyro replied:
This would fail if the key doesn't have a value. Like this:
![screenshot-repl it-2016-12-26-09-20-59](https://cloud.githubusercontent.com/assets/8657959/21482607/d910a3d6-cb4c-11e6-97f5-9cfdd7c25afb.png)
So what do you think if we also change the string for the keys that won't have replacements to end with a colon, and have an empty right side: `['key1:val1', 'key2:val2', 'key3:']`? It will evaluate as a key with an empty value (which is nice because is a falsy value).
![screenshot-repl it-2016-12-26-09-34-47](https://cloud.githubusercontent.com/assets/8657959/21482765/93463986-cb4e-11e6-82b0-1e86fc89437c.png)

moylop260 commented on pylint_odoo/checkers/no_modules.py:
```diff
@@ -327,10 +348,29 @@ class NoModuleChecker(BaseChecker):
+
+    def colon_list_to_dict(self, colon_list=None):
+        """ \
+        Converts a colon list to a dictionary
```
Use Sphinx docstring style:
```bash
:param colon_list list: Items strings with the strcuture... (your docstring for this field).
:returns: {'colon': 'list', 'just_key': None}
:rtype: dict
```

```

moylop260 commented on pylint_odoo/checkers/no_modules.py:
```diff
@@ -359,6 +399,15 @@ def visit_call(self, node):
                         # The argument "selection_add" is for overwrite field.
                         # Then don't need help.
                         has_help = None
+                    elif argument.arg in \
```
Avoid use `\`
You could use a `elif argument.arg in (\nself.config.deprecated_field_parameters):` instead.

moylop260 commented:
Show me the output running this check for addons-vauxoo too please

tomeyro commented:
I'm running into a bug @moylop260, but it's not part of my code. An exception is rising because the code is encountering a boolean where a string should be.

![captura de pantalla de 2016-12-26 11-05-44](https://cloud.githubusercontent.com/assets/8657959/21483957/31818fda-cb61-11e6-8351-6fcadb5c8639.png)
